### PR TITLE
[RLlib] Speedup A3C up to 3x (new `training_iteration` function instead of `execution_plan`) and re-instate Pong learning test.

### PR DESCRIPTION
**Pull request by sven1977**
This PR:
* Provides a new `training_iteration` function for A3C (alternative to existing `execution_plan`).
* By default, uses that new iteration function (`_disable_execution_plan_api=True`).
* ~3x speedup for `tuned_examples/a3c/pong-a3c.yaml` (16 worker, LSTM+CNN Atari problem).
* As a consequence of this speedup, A3C learns the Pong problem again with an LSTM -> re-instates previously out-commented weekly learning test case (similar to `tuned_examples/a3c/pong-a3c.yaml`).

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/a3c/a3c.py [hunk 1]
```diff
@@ -92,3 +110,60 @@ def execution_plan(
...
+            self._counters[NUM_AGENT_STEPS_SAMPLED] += result["agent_steps"]
+            self._counters[NUM_ENV_STEPS_SAMPLED] += result["env_steps"]
+            # Create current global vars.
+            global_vars = {
+                "timestep": self._counters[NUM_AGENT_STEPS_SAMPLED],
+            }
+
+            # Synch updated weights back to the particular worker.
+            with self._timers[SYNCH_WORKER_WEIGHTS_TIMER]:
+                weights = local_worker.get_weights(local_worker.get_policies_to_train())
```
Comment (avnishn) [inline rllib/agents/a3c/a3c.py:161]: Nice

File: rllib/agents/a3c/a3c.py [hunk 2]
```diff
@@ -92,3 +110,60 @@ def execution_plan(
...
+                worker.set_weights.remote(weights, global_vars)
+
+        # Update global vars of the local worker.
+        if global_vars:
+            local_worker.set_global_vars(global_vars)
+
+        # TODO: If we have processed more than one gradients
```
Comment (avnishn) [inline rllib/agents/a3c/a3c.py:168]: so to be clear we haven't written to result in this pr, right? But we want to for logging purposes.

**Review by avnishn — APPROVED** [review]

**Comment by avnishn** [comment]
Merge pending results dict.

File: rllib/agents/a3c/a3c.py (hunk 2 above)
Comment (sven1977) [reply rllib/agents/a3c/a3c.py:168]: This is still WIP. I need to add proper compilation of the results dict. The only thing that's missing is to combine those learner stats from all workers that have returned something from the `async_parallel_requests` call further above. This shim implementation right now only returns the last one.
Let me finish this before merging, of course.

**Pull request merged**